### PR TITLE
nav2_core and nav2_controller short documentation

### PR DESCRIPTION
**Pull request by yathartha3**
## Basic Info
Added a short documentation for nav2_controller, and nav2_core packages since it has none right now.

### CODE REVIEW

File: nav2_controller/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Nav2 Controller
+
+The Nav2 Controller is an [Execution Module](../doc/requirements/requirements.md) that implements the `nav2_behavior_tree::FollowPathAction` interface.
```
Comment (orduno) [inline nav2_controller/README.md]: Maybe remove the reference to behavior tree and intead mention it implements a `nav2_msgs::action::FollowPath` action server.

File: nav2_controller/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+An execution module implementing the `nav2_behavior_tree::FollowPathAction` interface is responsible for generating command velocities for the robot, given the computed path from the planner module in `nav2_planner`. The nav2_controller package is designed to be loaded with plugins for path execution. The plugins need to implement functions in the virtual base class defined in the `local_planner` header file in `nav2_core` package.
```
Comment (orduno) [inline nav2_controller/README.md]: An execution module implementing the `nav2_msgs::action::FollowPath` action server ...?

File: nav2_controller/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+
+Currently available controller plugins are: DWB, and [TEB (dashing release)](https://github.com/rst-tu-dortmund/teb_local_planner/tree/dashing-devel).
+
+
+The nav2_controller node is a Lifecycle node.
```
Comment (orduno) [inline nav2_controller/README.md]: `nav2_controller` is the namespace while `ControllerServer` is the node.

Comment (yathartha3) [reply nav2_controller/README.md]: I am removing this line. Maybe somewhere we should state what all nodes are lifecycle enabled.

File: nav2_controller/README.md (hunk 2 above)
Comment (orduno) [reply nav2_controller/README.md]: Maybe move reference to plugins to a a separate paragraph/section and add information about the `ControllerServer` and plugin roles.

File: nav2_controller/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Nav2 Controller
+
+The Nav2 Controller is an [Execution Module](../doc/requirements/requirements.md) that implements the `nav2_msgs::action::FollowPath` action server.
+
+An execution module implementing the `nav2_msgs::action::FollowPath` action server is responsible for generating command velocities for the robot, given the computed path from the planner module in `nav2_planner`. The nav2_controller package is designed to be loaded with plugins for path execution. The plugins need to implement functions in the virtual base class defined in the `local_planner` header file in `nav2_core` package.
+
+
+Currently available controller plugins are: DWB, and [TEB (dashing release)](https://github.com/rst-tu-dortmund/teb_local_planner/tree/dashing-devel).
```
Comment (SteveMacenski) [inline nav2_controller/README.md]: This is a type of comment that will keep growing as new options are around and other people want to get added. Id just say that you include Dwb and others can be found in the community

**Pull request merged**